zathura/utils.c

```diff
@@ -125,6 +128,82 @@ static gboolean search_current_index(GtkTreeModel* model, GtkTreePath* UNUSED(pa
   return FALSE;
 }
 
+static bool find_substring(const char* source_str, const char* search_str) {
+  gchar *normalized_source_str, *normalized_search_str;
+  gchar* case_normalized_source_str = NULL;
+  gchar* case_normalized_search_str = NULL;
+
+  normalized_source_str = g_utf8_normalize(source_str, -1, G_NORMALIZE_ALL);
+  normalized_search_str = g_utf8_normalize(search_str, -1, G_NORMALIZE_ALL);
+
+  if (normalized_search_str && normalized_source_str) {
+    case_normalized_source_str = g_utf8_casefold(normalized_source_str, -1);
+    case_normalized_search_str = g_utf8_casefold(normalized_search_str, -1);
+    return (strstr(case_normalized_source_str, case_normalized_search_str) != NULL);
+  }
+  return false;
+}
+
+static gboolean search_equal_iter(GtkTreeModel* model, gint column, const gchar* key, GtkTreeIter* iter) {
+
+  const gchar* source_string;
+  GValue value       = G_VALUE_INIT;
+  GValue transformed = G_VALUE_INIT;
+
+  gtk_tree_model_get_value(model, iter, column, &value);
+
+  g_value_init(&transformed, G_TYPE_STRING);
+
+  if (!g_value_transform(&value, &transformed)) {
+    g_value_unset(&value);
+    return TRUE;
+  }
+
+  g_value_unset(&value);
+  source_string = g_value_get_string(&transformed);
+  if (!source_string) {
+    g_value_unset(&transformed);
+    return TRUE;
+  }
+  if (find_substring(source_string, key)) {
+    return FALSE;
+  }
+  return TRUE;
+}
+
+gboolean search_equal_func_index(GtkTreeModel* model, gint column, const gchar* key, GtkTreeIter* iter,
+                                 gpointer search_data) {
```

Comment:
Reimplement the search equal function so that it will also compared with collapsed entries